internal/github/profile_test.go

```diff
@@ -0,0 +1,330 @@
+	key := generateKey(t)
+
+	// Example of an invalid profile URL
+	configURL := "github.com/chinmina/chinmina-bridge/docs/profile.yaml"
+	gh, err := github.New(
+		context.Background(),
+		config.GithubConfig{
+			ApiURL:         svr.URL,
+			PrivateKey:     key,
+			ApplicationID:  10,
+			InstallationID: 20,
+		},
+	)
+	require.NoError(t, err)
+
+	// Load the profile
+	_, err = github.GetProfile(context.Background(), gh, configURL)
+	require.Error(t, err)
+	assert.ErrorContains(t, err, ": 418")
+}
+
+// Test that the profile that is loaded is valid
+func TestValidProfile(t *testing.T) {
+
+	_, err := github.ValidateProfile(context.Background(), profile)
+
+	require.NoError(t, err)
+}
+
+// Test case where the profile that is loaded is invalid
+func TestInvalidProfile(t *testing.T) {
+
+	_, err := github.ValidateProfile(context.Background(), invalidProfile)
+
+	require.Error(t, err)
+
+}
+
+func TestLoadProfile(t *testing.T) {
+	router := http.NewServeMux()
+
+	expectedExpiry := time.Date(1980, 01, 01, 0, 0, 0, 0, time.UTC)
+
+	router.HandleFunc("/app/installations/{installationID}/access_tokens", func(w http.ResponseWriter, r *http.Request) {
+
+		JSON(w, &api.InstallationToken{
+			Token:     api.String("expected-token"),
+			ExpiresAt: &api.Timestamp{Time: expectedExpiry},
+		})
+	})
+
+	router.HandleFunc("/repos/chinmina/chinmina-bridge/contents/docs/profile.yaml", func(w http.ResponseWriter, r *http.Request) {
+
+		JSON(w, &api.RepositoryContent{
+			Content: &profile,
+		})
+	})
+
+	router.HandleFunc("/repos/chinmina/chinmina-bridge/contents/docs/invalid-profile.yaml", func(w http.ResponseWriter, r *http.Request) {
+
+		JSON(w, &api.RepositoryContent{
+			Content: &invalidProfile,
+		})
+	})
+
+	router.HandleFunc("/repos/chinmina/non-existent-profile.yaml", func(w http.ResponseWriter, r *http.Request) {
+
+		w.WriteHeader(http.StatusTeapot)
+	})
+
+	svr := httptest.NewServer(router)
+	defer svr.Close()
+
+	// generate valid key for testing
+	key := generateKey(t)
+
+	// Example of a valid profile URL
+	gh, err := github.New(
+		context.Background(),
+		config.GithubConfig{
+			ApiURL:         svr.URL,
+			PrivateKey:     key,
+			ApplicationID:  10,
+			InstallationID: 20,
+		},
+	)
+	require.NoError(t, err)
+
+	validProfile, _ := github.ValidateProfile(context.Background(), profile)
+
+	testCases := []struct {
+		configURL      string
+		expectedConfig github.ProfileConfig
+	}{
+		{
+			configURL:      "github.com/chinmina/chinmina-bridge/docs/profile.yaml",
+			expectedConfig: validProfile,
+		},
+		{
+			configURL:      "github.com/chinmina/non-existent-profile.yaml",
+			expectedConfig: github.ProfileConfig{},
+		},
+		{
+			configURL:      "github.com/chinmina/chinmina-bridge/docs/invalid-profile.yaml",
+			expectedConfig: github.ProfileConfig{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.configURL, func(t *testing.T) {
+			result, _ := github.LoadProfile(context.Background(), gh, tc.configURL)
```

Comment:
Ignoring the error here isn't quite right: at the very least we want to check that the error is in the expected state